src/scripts/components/VariableMessage.jsx

```diff
@@ -316,9 +366,55 @@ class VariableMessage extends React.Component {
    */
   showReset = () => this.props.reset && this.props.initialValue && (this.props.initialValue !== this.state.message);
 
+  onVariableDragStart = (event, variable) => {
+    const data = this.transformVar(variable);
+    this.currentDraggedSpanVariable = variable;
+    if (event.persist) event.persist();
+    event.dataTransfer.setData('text/html', data.outerHTML);
+    event.dataTransfer.setData('variableId', variable.id);
+  }
+
+  composeMessageDropHandler = (event) => {
+    const variableId = event.dataTransfer.getData('variableId');
+    setTimeout(() => {
+      const text = document.getElementById(`span-${this.currentDraggedSpanVariable.value}`);
```

Comment:
Are we able to use a ref here? Unfortunately Ids are not reliable in React when a component is used more than once on a page and can lead to unexpected bugs in the future.

Reply:
I think dealing with refs here would be difficult. As we don't know list length of variable props are coming from parent component as we have create ref's for all (all variables should be draggable), then we deal it. And also `transformVar()` function creates elements with basic js `document`. 

Maybe Dynamic Refs work. `one ref for all draggable <span's>`, But for that we have to change `transformVar()` that would be big change.